### DISCUSSION TITLE
GitHub Copilot free plan stopped working - rate limit exceeded

### DISCUSSION THREAD

**Discussion by giswqs** [community/community#43673] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I am a verified teacher with free access to Copilot. 

I keep getting the error `rate limit exceeded for plan FREE_FACULTY` today. It was working fine yesterday. Is there a solution to this? Thanks. 

> [ERROR] [fetch] [2023-01-09T03:53:58.117Z] Unhandled status from server:,429,{"error":{"message":"rate limit exceeded","internal_message":"rate limit exceeded for plan FREE_FACULTY","type":"client_error"}}

**Comment by otkrickey** [comment]
~Solved: #43634~

**Reply by giswqs** [reply]
This seems a different issue than https://github.com/orgs/community/discussions/43634

**Reply by otkrickey** [reply]
Sorry
Its my mistake
If you put the word "Teacher" in the title, it may be easier for the viewer to distinguish between them.

**Reply by JasinRefiku** [reply]
> Sorry Its my mistake If you put the word "Teacher" in the title, it may be easier for the viewer to distinguish between them.

It doesn't even matter if Teacher is or isn't written in the title, it's a totally different issue. Rate limited != Not working

**Comment by thakkarparth007** [comment]
Same issue, but for students: "rate limit exceeded for plan FREE_EDUCATIONAL".

**Reply by softwaredevelop** [reply]
I don't know solution but patience.. VS Code Copilot extensions have 3208246 users. I see the Light.
![copilot3](https://user-images.githubusercontent.com/61334390/211370638-fae76d1d-4109-46a1-94eb-a8fccb87f26e.png)

**Comment by IO-YO** [comment]
Same issue, also for students. I've programmed for three days straight now and it still works, but it doesn't work as smoothly as before. I have to sometimes redo comments over and over before copilot works.

Does anyone know if there is a rate limit?

```
[ERROR] [fetch] [2023-01-09T12:33:12.981Z] Unhandled status from server:,429,{"error":{"message":"rate limit exceeded","internal_message":"rate limit exceeded for plan FREE_EDUCATIONAL","type":"client_error"}}

[INFO] [default] [2023-01-09T12:33:13.834Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2023-01-09T12:33:14.055Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [ghostText] [2023-01-09T12:33:14.222Z] Cancelled during debounce
[INFO] [default] [2023-01-09T12:33:14.322Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2023-01-09T12:33:14.662Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 339 ms
[ERROR] [fetch] [2023-01-09T12:33:14.663Z] Unhandled status from server:,429,{"error":{"message":"rate limit exceeded","internal_message":"rate limit exceeded for plan FREE_EDUCATIONAL","type":"client_error"}}
```

**Reply by ciobiano** [reply]
Mine works but not as before , I really don’t know if it’s based on usage or something

**Comment by TiagoRibeiro25** [comment]
Getting this problem too, I've been using copilot for almost half a year and never had issues before

**Reply by Ahmadaljoudi** [reply]
same, still got it today :(

**Comment by Sculas** [comment]
I have this issue as well, but with the `FREE_ENGAGED_OSS` plan.

**Reply by wycats** [reply]
Likewise! I wouldn't mind paying for Copilot if this limitation is specific to auto-comped accounts.

**Comment by dav-ell** [comment]
Getting the `rate limit exceeded for plan FREE_FACULTY` error as well. IMO this limit should be raised for faculty, it happens after pressing ctrl+enter just a couple of times.

**Comment by Araxeus** [comment]
I think this might be a bug. I got this error after a few minutes of coding and then after a few minutes it started working again

**Comment by justlevine** [comment]
Don't think this is limited to teacher/student plans, my OSS plan is also failing
![image](https://user-images.githubusercontent.com/29322304/211415399-0cf694aa-6665-4aa0-b851-7b32e6f8279b.png)

**Comment by masands** [comment]
Same here
```
[INFO] [default] [2023-01-10T03:38:38.501Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2023-01-10T03:38:38.801Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2023-01-10T03:38:39.023Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2023-01-10T03:38:39.266Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 243 ms
[ERROR] [fetch] [2023-01-10T03:38:39.269Z] Unhandled status from server:,429,{"error":{"message":"rate limit exceeded","internal_message":"rate limit exceeded for plan FREE_EDUCATIONAL","type":"client_error"}}
```

**Comment by fahim9778** [comment]
I am having the same issue since last night:
`[ERROR] [fetch] [2023-01-10T05:44:58.320Z] Unhandled status from server:,429,{"error":{"message":"rate limit exceeded","internal_message":"rate limit exceeded for plan FREE_EDUCATIONAL","type":"client_error"}}`

I have never faced this issue during the last 6 months. Can any @github-staff help us with this issue?

**Reply by Ahmadaljoudi** [reply]
same here, let me know if you found anything

**Comment by susanforme** [comment]
me too

**Comment by Duoquote** [comment]
Looks like the rate limit is reduced on server side but now it makes the service unusable, you can not get consecutive recommendations...

**Comment by Joshua-Noakes1** [comment]
Same issue here, been happening since yesterday

**Reply by MarcusSlover** [reply]
Same here

**Reply by Ahmadaljoudi** [reply]
same same

**Comment by mendax0110** [comment]
Same issue here

**Comment by zjeffer** [comment]
Getting the same thing. My plan is `FREE_ENGAGED_OSS`. Started happening yesterday. Makes the thing pretty useless if it only works 1/10 times.

**Reply by TiagoRibeiro25** [reply]
There's no point using this if it barely works

**Reply by zjeffer** [reply]
I was actually going to buy it until I noticed I had free access, but now I **can't** buy it because there's no button to buy it if you have free access. 

But as far as I can tell this rate limit only affects the free plans, so just because I'm a maintainer for a popular open source repo I basically have a worse experience with no option to upgrade? wtf?

**Reply by TiagoRibeiro25** [reply]
There shouldn't be a difference between both unless they changed that yesterday.

**Reply by zjeffer** [reply]
Github is the opposite of transparent regarding Copilot so I'm guessing they silently changed it to give preference to people who bought it or something.

**Comment by psapezhka** [comment]
The same issue here

**Comment by fuxiaohei** [comment]
The same issue. 
No **Edit** button in `settings/Billing and plans` page but in https://docs.github.com/en/billing/managing-billing-for-github-copilot/managing-your-github-copilot-for-individuals-subscription. 
Just show `Thank you for participating in the GitHub Copilot Technical Preview. Your extended access has ended.`
 
![image](https://user-images.githubusercontent.com/2142787/211571304-a7270f36-edd9-4a89-a48d-d2b108006543.png)

no way to buy a plan.

**Reply by theblackturtle** [reply]
Me too

**Comment by MarcusSlover** [comment]
Same issue here.
`{"error":{"message":"rate limit exceeded","internal_message":"rate limit exceeded for plan FREE_EDUCATIONAL","type":"client_error"}}`

**Comment by Ahmadaljoudi** [comment]
same here :(

**Comment by denizyoldas** [comment]
same problem

**Comment by MarcoSpanoDev** [comment]
Sadly same here :( I miss my Copilot

**Comment by Rocamonde** [comment]
Same here, this makes the product utility drop by 90%; need to wait 4-5 seconds before triggering the next autocomplete, which is approximately the time it takes for me to think of a solution a lot of the time. So effectively cannot be used to reduce cognitive overload on simple tasks, which is where most of the value of Copilot comes from, since for anything relatively complex or involved I cannot get a zero-shot correct completion and have to end up designing the high level pattern myself and reducing it to the kinds of tasks I mentioned above.

**Comment by MarcoSpanoDev** [comment]
What helps for me is pressing backspace and space again after a couple of times it will autocorrect for some lines again. Not amazing but hey its something

**Reply by Rocamonde** [reply]
yeah, that's what I do. or backspace-enter if you're on a newline. But this takes too long.

**Comment by raulibanez** [comment]
Same, I've got this on Github settings webpage.

"Thank you for participating in the GitHub Copilot Technical Preview. Your extended access has ended."

**Comment by soldni** [comment]
I would happily switch to a paid subscription, but the settings UI gives me no option to sign up:
![Screenshot 2023-01-10 at 14 35 41](https://user-images.githubusercontent.com/3913506/211676995-9aa0acf6-f331-4ed1-8a9c-aecb4278e6aa.png)

**Reply by h8rry** [reply]
Yeah is there a way tp upgrade to paid instead?

**Reply by kachkaev** [reply]
Same here since very recently. I’ve been granted a free license as a maintainer of Prettier.

<img width="795" alt="Screenshot 2023-01-12 at 12 47 08" src="https://user-images.githubusercontent.com/608862/212070369-ba6f974f-f334-499d-b495-d1f95450c890.png">
<img width="819" alt="Screenshot 2023-01-12 at 12 47 15" src="https://user-images.githubusercontent.com/608862/212070374-59a1fe22-1bb2-4b99-82f7-2a213378339d.png">

**Comment by Lioness100** [comment]
Same issue as a student.

**Comment by lrdcxdes** [comment]
+

**Comment by fahim9778** [comment, marked as answer]
I got an answer from the GitHub support team.

I hope this becomes true. 🤲🙏
![Screenshot_20230111_181212_Chrome](https://user-images.githubusercontent.com/26436023/211805265-aad2fb59-9857-4e5b-8856-9849e053a88c.jpg)

**Reply by zjeffer** [reply]
Thanks for this! I guess this issue can be closed, as we know the reason, now? @giswqs

**Reply by fahim9778** [reply]
> Thanks for this! I guess this issue can be closed, as we know the reason, now? @giswqs

I think let's keep it open till at least till 16th. If it seems okay from then, we can close.
@giswqs @zjeffer

**Reply by zjeffer** [reply]
I meant that he could mark your comment as an answer, so that people who visit this thread for the first time immediately see what the reason is :)

**Reply by fahim9778** [reply]
> I meant that he could mark your comment as an answer, so that people who visit this thread for the first time immediately see what the reason is :)

Okay!

**Reply by giswqs** [reply]
Answer marked. Thanks for sharing

**Reply by Rocamonde** [reply]
Microsoft is clearly struggling to scale up OpenAI's models -- ChatGPT constantly at capacity, Codex is too busy...

**Reply by ImJustChew** [reply]
They should at least send a notification email detailing this.

**Reply by ReaVNaiL** [reply]
An official statement could've helped. Something like: down this week for maintenance, I mean just for user awareness.

**Reply by TiagoRibeiro25** [reply]
> They should at least send a notification email detailing this.

I totally agree with that, keeping us in the dark is kinda lame

**Reply by Veyg** [reply]
Thanks for the answer! 

Please don't forget to forward this answer to other people, as you discussed above, there was no official statement neither an notification

**Reply by wycats** [reply]
As I said up-thread, if GitHub operations folks think that this sort of tradeoff makes sense, I'd **really** prefer to pay for Copilot and never have to worry about it.

**Reply by th3cskid** [reply]
agreed, should have gotten some sort of statement! 

i follow the guy who reverse engineered copilot, and he is suggesting using codeium (https://twitter.com/parth007_96/status/1613300315658551298). anyone else tried it?

**Comment by leng-yue** [comment]
Why it says that `rate limit exceeded for plan FREE_EDUCATIONAL` even if I already paid for it?

<img width="728" alt="image" src="https://user-images.githubusercontent.com/25119060/211920006-54359feb-4f51-449c-970e-a75a4474dac9.png">

**Comment by DavidVidaurre** [comment]
Even after receiving confirmation of my renewal, the copilot does not work.
![image](https://user-images.githubusercontent.com/89712142/211972833-83d73577-7f25-4681-9264-5434b3932a63.png)

**Comment by CipherLab** [comment]
I'm paying for copilot and I'm getting: 
2024-01-12 10:55:46.893 [error] [Extension Host] github.copilot.signIn Error: API rate limit exceeded for user ID 3710501. If you reach out to GitHub Support for help, please include the request ID DDC3:8444:400535D:8481447:65A16F3C.

**Comment by dh-saurabh** [comment]
me too I register for the yearly plan and I am getting
```Sorry, your request was rate-limited. Please wait and try sending again later.```

**Comment by miljoc** [comment]
Here too, guess they have a service disruption

**Comment by vrogers** [comment]
yep, I'm getting the same issue as a paid user.

**Comment by davecheney** [comment]
Duplicate of https://github.com/orgs/community/discussions/43865
